components/raftstore-v2/src/operation/command/admin/split.rs

```diff
@@ -64,20 +74,387 @@ pub struct SplitResult {
+            panic!("{:?} original region should exist", self.logger.list());
+        }
+        let last_region_id = regions.last().unwrap().get_id();
+        for (new_region, locks) in regions.into_iter().zip(region_locks) {
+            let new_region_id = new_region.get_id();
+
+            if new_region_id == region_id {
+                let not_exist = meta
+                    .region_ranges
+                    .insert(enc_end_key(&new_region), new_region_id)
+                    .is_none();
+                assert!(not_exist, "[region {}] should not exist", new_region_id);
+                continue;
+            }
+
+            let mut raft_message = self.prepare_raft_message();
+            raft_message.mut_from_peer().set_id(raft::INVALID_ID);
+            raft_message.set_region_id(new_region_id);
+            raft_message.set_to_peer(
+                new_region
+                    .get_peers()
+                    .iter()
+                    .find(|p| p.store_id == store_ctx.store_id)
+                    .unwrap()
+                    .clone(),
+            );
+
+            self.split_progress_mut().insert(new_region_id, false);
+            let init_info = SplitRegionInitInfo {
+                region: new_region,
+                parent_region_id: derived_id,
+                parent_epoch: derived_epoch.clone(),
+                parent_is_leader: self.is_leader(),
+                parent_stat: self.peer_stat().clone(),
+                approximate_keys: estimated_keys,
+                approximate_size: estimated_size,
+                locks,
+                last_split_region: last_region_id == new_region_id,
+            };
+
+            store_ctx
+                .router
+                .send_control(StoreMsg::CreatePeer(Box::new(CreatePeer {
+                    raft_message,
+                    split_region_info: init_info,
+                })));
+        }
+        drop(meta);
+
+        if is_leader {
+            self.on_split_region_check_tick();
+        }
+    }
+
+    pub fn init_split_region<T>(
+        &mut self,
+        store_ctx: &mut StoreContext<EK, ER, T>,
+        init_info: Box<SplitRegionInitInfo>,
+    ) {
+        let SplitRegionInitInfo {
+            parent_region_id,
+            parent_epoch,
+            region,
+            parent_is_leader,
+            parent_stat,
+            approximate_keys,
+            approximate_size,
+            locks,
+            last_split_region,
+        } = Box::into_inner(init_info);
+
+        let mut need_schedule_apply_fsm = false;
+        let region_id = region.id;
+
+        let replace = region.get_region_epoch().get_version()
+            > self
+                .storage()
+                .region_state()
+                .get_region()
+                .get_region_epoch()
+                .get_version();
+
+        if !self.storage().is_initialized() || replace {
+            let mut wb = store_ctx.engine.log_batch(5);
```

Comment:
use the const instead

Reply:
It's the capacity of the log batch, rather than the log index.

Reply:
> It's the capacity of the log batch, rather than the log index.

Then it might be better to add a small comment why it's 5.

Reply:
As this PR does not consider persistency now, these codes have been removed.